src/hotspot/share/gc/g1/g1CollectionSet.cpp

```diff
@@ -99,12 +99,21 @@ void G1CollectionSet::initialize(uint max_region_length) {
   _candidates.initialize(max_region_length);
 }
 
+void G1CollectionSet::abandon() {
+  _g1h->young_regions_cset_group()->clear(true /* uninstall_cset_group */);
```

Comment:
It's a bit odd that `CollectionSet` doesn't have direct access to `cset_group` and have to go through heap.

Reply:
I thought back and forth making `G1CollectionSet` the owner of the `young_regions_cset_group` (i.e. a member of it) and found both advantages and disadvantages for that.
Unless the second reviewer thinks it is better to do that, I'll keep it for now.